src/Moryx.AbstractionLayer.Products.Endpoints/ProductManagementController.cs

```diff
@@ -0,0 +1,332 @@
+                   .FirstOrDefault();
+            if (type == null)
+                return NotFound();
+            var productType = (ProductType)Activator.CreateInstance(type);
+            var newType = _productConverter.ConvertProductBack(newTypeModel, productType);
+            return _productManagement.SaveType(newType);
+        }
+
+        [HttpPost]
+        [Route("producttypes/query")]
+        public ProductModel[] GetTypes(ProductQuery query)
+        {
+            var productTypes = _productManagement.LoadTypes(query);
+            var productModels = new List<ProductModel>();           
+            foreach (var t in productTypes)
+            {
+                productModels.Add(_productConverter.ConvertProduct(t, false));
+            }
+            return productModels.ToArray();
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{id}")]
+        public ActionResult<ProductModel> GetType(long id)
+        {
+            if (id == 0)
+                return BadRequest($"Id was 0");
+            var productType = _productManagement.LoadType(id);
+            if (productType == null)
+                return NotFound();
+            return _productConverter.ConvertProduct(productType, false);
+        }
+
+        [HttpDelete]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{id}")]
+        public ActionResult<bool> DeleteType(long id)
+        {
+            var result = _productManagement.DeleteProduct(id);
+            if (!result)
+                return NotFound();
+            return result;
+        }
+
+        [HttpPut]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Route("producttypes/{id}")]
+        public ActionResult<long> UpdateType(long id, ProductModel modifiedType)
+        {
+            if (modifiedType == null)
+                return BadRequest($"Modified product type was null");
+            var type = _productManagement.LoadType(id);
+            if (type == null)
+                return BadRequest($"No product type with id {modifiedType.Id} was found");
+            type = _productConverter.ConvertProductBack(modifiedType, (ProductType) type);
+            return _productManagement.SaveType(type);
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{identifier}/{revision}")]
+        public ActionResult<ProductModel> GetType(string identifier, short revision)
```

Comment:
The identifier after types is mixed between id and identifier. This is inconsistent.